tests/common/configuration/node_config_model.rs

```diff
@@ -72,25 +72,33 @@ impl NodeConfig {
         }
     }
 
+    pub fn regenerate_ports(&mut self) {
+        self.rest.listen = format!("127.0.0.1:{}", NodeConfig::get_available_port().to_string());
+        self.peer_2_peer.public_address = format!(
+            "/ip4/127.0.0.1/tcp/{}",
+            NodeConfig::get_available_port().to_string()
+        );
+    }
+
     pub fn get_node_address(&self) -> String {
         let output = format!("http://{}/{}", self.rest.listen, self.rest.prefix);
         output
     }
-}
 
-fn get_available_port() -> u16 {
-    let available_port = loop {
-        let port = rand::thread_rng().gen_range(8000, 9999);
-        if port_is_available(port) {
-            break port;
-        }
-    };
-    available_port
-}
+    fn get_available_port() -> u16 {
+        let available_port = loop {
+            let port = rand::thread_rng().gen_range(8000, 9999);
+            if NodeConfig::port_is_available(port) {
+                break port;
+            }
+        };
+        available_port
+    }
 
-fn port_is_available(port: u16) -> bool {
-    match TcpListener::bind(("127.0.0.1", port)) {
-        Ok(_) => true,
-        Err(_) => false,
+    fn port_is_available(port: u16) -> bool {
+        match TcpListener::bind(("127.0.0.1", port)) {
```

Comment:
If the listener is dropped immediately, the port is released, so there is a race condition.

Reply:
I changed it to bind ot 0 port which binds to random unoccupied port.

Reply:
What I meant is that we can change jormungandr to bind gRPC to a random port. The problem is how to discover that port in a test after starting jormungandr.

Another way is to hand an open, pre-bound socket descriptor down to jormungandr with a particular option. IIRC this is also how systemd loves to start daemons on demand. But I only know how to do this on Unix.